radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/application/TimestampChecker.java

```diff
@@ -62,7 +63,8 @@ public TimestampChecker(Duration acceptableTimeRange) {
 	public Observable<TestInvariantError> check(RunningNetwork network) {
 		return network.ledgerUpdates()
 			.map(Pair::getSecond)
-			.distinct() // Test on only the first ledger update in the network
+			// Test on only the first ledger update in the network
+			.distinct(update -> EpochView.of(update.getTail().getEpoch(), update.getTail().getView()))
 			.filter(l -> !(l.getTail().getEpoch() == 1 && l.getTail().getView().equals(View.of(1))))
```

Comment:
Probably it worth to move filtering condition into dedicated method with descriptive name.

Reply:
good idea